stl/inc/complex

```diff
@@ -1845,49 +1845,28 @@ _NODISCARD complex<_Upgrade_to_double<_Ty>> proj(_Ty _Left) {
 // FUNCTION TEMPLATE pow
 template <class _Ty1, class _Ty2>
 _NODISCARD complex<_Common_float_type_t<_Ty1, _Ty2>> pow(const complex<_Ty1>& _Left, const complex<_Ty2>& _Right) {
-    using type = complex<_Common_float_type_t<_Ty1, _Ty2>>;
-    return _STD pow(type(_Left), type(_Right));
+    using _Type = complex<_Common_float_type_t<_Ty1, _Ty2>>;
+    return _STD pow(_Type(_Left), _Type(_Right));
 }
 
-template <class _Ty1, class _Ty2, enable_if_t<!is_integral_v<_Ty2>, int> = 0>
+template <class _Ty1, class _Ty2, enable_if_t<is_arithmetic_v<_Ty2>, int> = 0>
 _NODISCARD complex<_Common_float_type_t<_Ty1, _Ty2>> pow(const complex<_Ty1>& _Left, const _Ty2& _Right) {
-    using type = complex<_Common_float_type_t<_Ty1, _Ty2>>;
-    return _STD pow(type(_Left), type(_Right));
-}
-
-template <class _Ty1, class _Ty2, enable_if_t<!is_integral_v<_Ty1> && is_integral_v<_Ty2>, int> = 0>
-_NODISCARD complex<_Common_float_type_t<_Ty1, _Ty2>> pow(const complex<_Ty1>& _Left, const _Ty2 _Right) {
-    using type = complex<_Common_float_type_t<_Ty1, _Ty2>>;
-
-    type _Tmp   = _Left;
-    auto _Count = static_cast<make_unsigned_t<_Ty2>>(_Right);
-
-    if (_Right < 0) {
-        _Count = 0 - _Count; // safe negation as unsigned
-    }
-
-    for (type _Zv(1);; _Tmp *= _Tmp) { // fold in _Left ^ (2 ^ _Count) as needed
-        if ((_Count & 1) != 0) {
-            _Zv *= _Tmp;
-        }
-
-        if ((_Count >>= 1) == 0) {
-            return _Right < 0 ? type(1) / _Zv : _Zv;
-        }
-    }
+    using _Promoted = _Common_float_type_t<_Ty1, _Ty2>;
+    using _Type     = complex<_Promoted>;
+    return _STD pow(_Type(_Left), _Type(static_cast<_Promoted>(_Right)));
 }
 
 template <class _Ty1, class _Ty2, enable_if_t<is_integral_v<_Ty1> && is_integral_v<_Ty2>, int> = 0>
 _NODISCARD complex<_Ty1> pow(const complex<_Ty1>& _Left, _Ty2 _Right) {
     // raise Gaussian integer to an integer power

```

Comment:
The change to the overload set (not the `_Type` renaming which is fine) has the potential to cause compiler errors for `pow()` being called with these non-Standard Gaussian integers. Example:

```
C:\Temp>type meow.cpp
```
```cpp
#include <complex>
#include <iostream>
using namespace std;

template <typename T>
void print(const complex<T>& c) {
    cout << "(" << c.real() << ", " << c.imag() << ")\n";
}

int main() {
    const complex<int> gaussian_integer{3, 4};
    print(pow(gaussian_integer, 2));
    print(pow(gaussian_integer, 2L));
}
```

Before, with VS 2019 16.8 Preview 3:
```
C:\Temp>cl /EHsc /nologo /W4 meow.cpp && meow
meow.cpp
(1, 0)
(-7, 24)
```

After this PR:

```
C:\Temp>cl /EHsc /nologo /W4 meow.cpp && meow
meow.cpp
meow.cpp(13): error C2668: 'pow': ambiguous call to overloaded function
S:\GitHub\STL\out\build\x86\out\inc\complex(1860): note: could be 'std::complex<int> std::pow<int,long,0>(const std::complex<int> &,_Ty2)'
        with
        [
            _Ty2=long
        ]
S:\GitHub\STL\out\build\x86\out\inc\complex(1853): note: or       'std::complex<double> std::pow<int,long,0>(const std::complex<int> &,const _Ty2 &)'
        with
        [
            _Ty2=long
        ]
meow.cpp(13): note: while trying to match the argument list '(const std::complex<int>, long)'
meow.cpp(13): error C2672: 'print': no matching overloaded function found
```

The `pow(gaussian_integer, 2)` line printed `(1, 0)` because it was invoking
https://github.com/microsoft/STL/blob/530bdc5aaa8a21277e1281ad3df8b8d8433b5caa/stl/inc/complex#L1621-L1622
and that was unprepared to handle Gaussian integers. (This overload was selected because saying `_Ty` twice makes it more specialized than overloads that permit `_Ty1` and `_Ty2` to vary.) This indicates that our support for Gaussian integers is really terrible (I knew it was bad after @statementreply discovered that `sqrt` crashes, but my foolish thinking was "surely `pow` is fine because we have all of those dedicated overloads").

Previously, `pow(gaussian_integer, 2L)` managed to print `(-7, 24)` because the three overloads here were crafted (deliberately during maintenance, but we failed to actually _test_ this, or we would have discovered the `complex<int>, int` issue above) to be unambiguous amongst themselves:

* `enable_if_t<!is_integral_v<_Ty2>, int> = 0>`
* `enable_if_t<!is_integral_v<_Ty1> && is_integral_v<_Ty2>, int> = 0>`
* `enable_if_t<is_integral_v<_Ty1> && is_integral_v<_Ty2>, int> = 0>`

In this PR, the remaining overloads are ambiguous for `complex<int>, long` (the signatures are effectively identical so only the SFINAE constraints can prevent ambiguity):
* `enable_if_t<is_arithmetic_v<_Ty2>, int> = 0>`
* `enable_if_t<is_integral_v<_Ty1> && is_integral_v<_Ty2>, int> = 0>`

After looking at this, I think we should simply remove this "// raise Gaussian integer to an integer power" overload - it never activated for `complex<int>, int`, yet nobody has complained in over a decade. I suspect that there are no users invoking it with `complex<int>, long` and similar varying types (if there are, we can always investigate restoring this logic in a fixed form, or they can extract this code into a differently named function).